js/tweek-client/src/TweekManagementClient/TweekManagementClient.ts

```diff
@@ -233,6 +234,58 @@ export default class TweekManagementClient implements ITweekManagementClient {
     return this._fetch(url).then(toJson);
   }
 
+  async getHooks(): Promise<HooksReponse> {
+    const url = `${this.config.baseServiceUrl}/api/v2/hooks`;
+    const res = await this._fetch(url);
+
+    return {
+      hooks: await res.json(),
+      etag: res.headers.get('ETag') as string,
+    };
+  }
+
+  async getHooksForKeyPath(keyPath: string): Promise<HooksReponse> {
+    const url = `${this.config.baseServiceUrl}/api/v2/hooks/${keyPath}`;
+    const res = await this._fetch(url);
+
+    return {
+      hooks: await res.json(),
+      etag: res.headers.get('ETag') as string,
+    };
+  }
+
+  createHook(etag: string, keyPath: string, type: string, url: string): Promise<void> {
+    const requestUrl = `${this.config.baseServiceUrl}/api/v2/hooks/${keyPath}`;
+    const config = {
+      method: 'POST',
+      headers: { 'If-Match': etag, ...jsonHeaders },
+      body: JSON.stringify({ type, url }),
+    };
+
+    return this._fetch(requestUrl, config).then(noop);
+  }
+
+  updateHook(etag: string, keyPath: string, hookIndex: number, type: string, url: string): Promise<void> {
```

Comment:
Hmm... to make the update hook simpler, maybe it's worth adding a method style:
```[update|delete}Hook(keyPath, ([hooks])-> index, type, url )```
That way, we can simplify the api.

Reply:
This method will automatically fetch and pass the etag

Reply:
I'm not sure I understood this well

If we fetch and pass the etag when we want to update/delete then we are not comparing it to the original etag that was in place when the user saw the hooks and decided on this update/delete, so we are not preventing the potentially wrong update

Unless you were talking about somehow combining the get request with the update/delete and somehow keeping the etag behind the scenes? that sounds like it would be more complicated

I was planning to save the etag somewhere in the hook page context and passing it back when doing the update/delete

Reply:
I can possibly keep the last etag received somewhere internally in the client and then send it along with update/delete requests, though that will only work for frontend.
If the client can be used from a server that might run multiple get/update commands concurrently this will behave erratically